File: 0506-relative-ranks/0506-relative-ranks.cpp
class Solution {
public:
    vector<string> findRelativeRanks(vector<int>& score) {
        int n = score.size();
        vector<string> ans(n);
        priority_queue<pair<int,int>> pq;
        for(int i = 0; i < n; ++i){
            pq.push({score[i], i});
        }
        for(int i = 1; i <= n; ++i){
            string rank = "";
            if(i == 1) rank = "Gold Medal";
            else if(i == 2) rank = "Silver Medal";
            else if(i == 3) rank = "Bronze Medal";
            else rank = to_string(i);
            ans[pq.top().second] = rank;
            pq.pop();
        }
        return ans;
    }
};

File: 0506-relative-ranks/0506-relative-ranks.cs
public class Solution {
    public string[] FindRelativeRanks(int[] score) {
        //  https://leetcode.com/problems/relative-ranks/solutions/3857694/c-solution-priorityqueue-o-nlogn
        //  https://learn.microsoft.com/ko-kr/dotnet/api/system.collections.generic.priorityqueue-2?view=net-8.0
        int n = score.Length;
        PriorityQueue<int, int> pq = new(); 
        string[] ans = new string[n];
        for(int index = 0; index < n; ++index){
            pq.Enqueue(index, -score[index]);
        }
        if(pq.Count > 0){
            ans[pq.Dequeue()] = "Gold Medal";
        }

        if(pq.Count > 0){
            ans[pq.Dequeue()] = "Silver Medal";
        }
        
        if(pq.Count > 0){
            ans[pq.Dequeue()] = "Bronze Medal";
        }
        
        for(int i=4; pq.Count > 0; i++){
            //  https://learn.microsoft.com/ko-kr/dotnet/csharp/language-reference/tokens/interpolated
            //  res[pq.Dequeue()] = $"{i}";
            ans[pq.Dequeue()] = i.ToString();
        }
        return ans;
    }
}